File: tools/bench/answers/python_haiku/s1/09_topwords.py
sentence = "the cat sat on the mat with the cat"
words = sentence.split()
counts = {}
for word in words:
    if word in counts:
        counts[word] += 1
    else:
        counts[word] = 1
sorted_words = sorted(counts.items(), key=lambda x: x[1], reverse=True)
for i in range(2):
    word, count = sorted_words[i]
    print(f"{word} {count}")

File: tools/bench/answers/go_haiku/s1/09_topwords.go
package main

import (
	"fmt"
	"sort"
	"strings"
)

func main() {
	s := "the cat sat on the mat with the cat"
	words := strings.Fields(s)
	counts := make(map[string]int)
	for _, word := range words {
		counts[word]++
	}
	type pair struct {
		word  string
		count int
	}
	var pairs []pair
	for word, count := range counts {
		pairs = append(pairs, pair{word, count})
	}
	sort.Slice(pairs, func(i, j int) bool {
		return pairs[i].count > pairs[j].count
	})
	for i := 0; i < 2 && i < len(pairs); i++ {
		fmt.Printf("%s %d\n", pairs[i].word, pairs[i].count)
	}
}
